ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp

```diff
@@ -369,7 +369,7 @@ Tensor _outer(const Tensor& input_a, const Tensor& input_b, const std::optional<
         }
     }
 
-    return ttnn::matmul(a_slim, b_slim);
+    return ttnn::matmul(a_slim, b_slim, /*transpose_a=*/false, /*transpose_b=*/false, /*memory_config=*/ttnn::DRAM_MEMORY_CONFIG, /*dtype=*/std::nullopt, /*program_config=*/std::nullopt, /*activation=*/std::nullopt, /*compute_kernel_config=*/std::nullopt, /*core_grid=*/std::nullopt);
```

Comment:
```suggestion
    return ttnn::matmul(a_slim, b_slim, /*transpose_a=*/false, /*transpose_b=*/false, /*memory_config=*/ttnn::DRAM_MEMORY_CONFIG;
```

and if you change transpose bools to enum flags

```
enum class Transpose : uint8_t {
    None   = 0,
    Node   = 1 << 0,
    InputA = 1 << 1,
    InputB = 1 << 2, 
    Both   = InputA | InputB // 0110
};
```

you can then have a default value for it to be None. In this case you can shorten the call to

```suggestion
    return ttnn::matmul(a_slim, b_slim, Transpose::None, ttnn::DRAM_MEMORY_CONFIG);
```

and if memory config becomes optional, than you can eliminate both 2 remaining arguments. 
when optional value is not passed we typically get the value from the first input.

Reply:
I don't insist for this change in this PR. But recommend to follow that path

Reply:
I followed the new ttnn operation example a bit too closely.

Once I passed in default values I could remove this code change.

I also changed memory_config to be optional. If the workflows pass I'll keep that change.